src/libostree/ostree-repo-checkout.c

```diff
@@ -102,48 +102,62 @@ fsync_is_enabled (OstreeRepo   *self,
 static gboolean
 write_regular_file_content (OstreeRepo            *self,
                             OstreeRepoCheckoutAtOptions *options,
-                            GOutputStream         *output,
+                            int                    outfd,
                             GFileInfo             *file_info,
                             GVariant              *xattrs,
                             GInputStream          *input,
                             GCancellable          *cancellable,
                             GError               **error)
 {
   const OstreeRepoCheckoutMode mode = options->mode;
+  g_autoptr(GOutputStream) outstream = NULL;
 
-  if (g_output_stream_splice (output, input, 0,
-                              cancellable, error) < 0)
-    return FALSE;
+  if (G_IS_FILE_DESCRIPTOR_BASED (input))
```

Comment:
We expose as public API:
```
gboolean      ostree_repo_write_content (OstreeRepo       *self,
                                         const char       *expected_checksum,
                                         GInputStream     *object_input,
                                         guint64           length,
                                         guchar          **out_csum,
                                         GCancellable     *cancellable,
                                         GError          **error);
```

So we have to support it.  In practice using the command line tools, we will go through this code path when pulling from an `archive` repo to a `bare` one, since the content objects are zlib-compressed.